test/test_automated_install.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from textwrap import dedent
+
+@pytest.fixture
+def Pihole(Docker):
+    ''' generates stubs of files that normally execute code '''
+    ''' (moved to Dockerfile in some cases) '''
+    return Docker
+
+SETUPVARS = {
+    'piholeInterface' : 'eth99',
+    'IPv4addr' : '192.168.100.2',
+    'piholeIPv6' : 'True',
+    'piholeDNS1' : '4.2.2.1',
+    'piholeDNS2' : '4.2.2.2'
+}
+
+def test_setupVars_are_sourced_to_global_scope(Pihole):
+    ''' one function imports, other functions read these variables '''
+    setup_var_file = 'cat <<EOF> /etc/pihole/setupVars.conf\n'
+    for k,v in SETUPVARS.iteritems():
+        setup_var_file += "{}={}\n".format(k, v)
+    setup_var_file += "EOF\n"
+    Pihole.run(setup_var_file).stdout
+
+    script = dedent('''#!/bin/bash -e
+    . /opt/pihole/stub_basic-install.sh
+    readSetupVarsIfPresent pihole
```

Comment:
Earlier I mentioned "functions to call at your leisure.", here is where I do that.